### PR TITLE
Warrior update

### PR DESCRIPTION
**Pull request by vicgal**
Update all layers to warrior

### CODE REVIEW

File: layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample [hunk 1]
```diff
@@ -14,7 +14,9 @@ BBLAYERS ?= " \
     ${TOPDIR}/../layers/meta-openembedded/meta-python \
     ${TOPDIR}/../layers/meta-sunxi \
     ${TOPDIR}/../layers/meta-balena/meta-resin-common \
```
Comment (floion) [inline layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample]: Please also change this to `meta-balena-common`

File: layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample [hunk 2]
```diff
@@ -14,7 +14,9 @@ BBLAYERS ?= " \
...
-    ${TOPDIR}/../layers/meta-balena/meta-resin-sumo \
+    ${TOPDIR}/../layers/meta-balena/meta-balena-warrior \
     ${TOPDIR}/../layers/meta-balena-allwinner \
     ${TOPDIR}/../layers/meta-rust \
     "
+
+BBMASK += "${TOPDIR}/../layers/meta-sunxi/recipes-bsp/u-boot/u-boot_%.bbappend"
```
Comment (floion) [inline layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample]: Why this change? We need to have it detailed in the git log why this was done

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11.bb [hunk 3]
```diff
@@ -63,6 +63,7 @@ SRC_URI = "git://git.denx.de/u-boot.git;branch=master \
         file://boot.cmd \
         file://0001-Increase-size-of-memory-available-to-bootm.patch \
         file://0002-Add-Resin-specific-boot-command.patch \
+	file://0003-libftd_fix.patch \
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11.bb]: Please fix indentation

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11/0003-libftd_fix.patch [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+Subject: [PATCH] Include U-Boot libfdt headers from their actual path
+
+There are no headers for libfdt in lib/libfdt, as they are instead
+located in scripts/dtc/libfdt. Specifying lib/libfdt for headers
+inclusion in host tools results in using the system libfdt headers,
+which is not what we want. Change this to the proper path.
+
+Signed-off-by:Paul Kocialkowski <contact@paulk.fr>
+
+Upstream-Status: Inappropriate [Not author]
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11/0003-libftd_fix.patch]: Is this already in upstream? If yes, then status should be instead `Backport`

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+DESCRIPTION="Upstream's U-boot configured for sunxi devices"
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb]: What u-boot version is the BSP layer using now? Why can't we use that one too?

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend [hunk 6]
```diff
@@ -127,3 +128,7 @@ RESIN_CONFIGS_append_orange-pi-lite = " \
 RESIN_CONFIGS[8189fs] ?= " \
     CONFIG_RTL8189FS=m \
 "
+
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: What is the build error?

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11/0003-libftd_fix.patch (hunk 4 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11/0003-libftd_fix.patch]: Done

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11.bb (hunk 3 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2017.11.bb]: Done

File: layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample (hunk 2 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample]: The masked file, if unmasked, applies to u-boot recipes from `meta-balena-allwiner` and this is not necessary as all the recipes files have proper/custom bbappends according to the u-boot version they refer to.
I will update the git log to reflect this.

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb (hunk 5 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb]: It's 2018.09 but that is failing when meta-balena tries to apply balena-specific-env patches.

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend (hunk 6 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: ERROR: linux-4.14.14+gitAUTOINC+9c0bf98471-r0 do_package: QA Issue: linux: Files/directories were installed but not shipped in any package:
  /boot/sun8i-h3-fixup.scr
Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb (hunk 5 above)
Comment (floion) [reply layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb]: Then those patches need to be redone for the new u-boot. I don't understand why we should pin u-boot to an older release rather than redo those patches.

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend (hunk 6 above)
Comment (floion) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: How come this new file got generated now? We did not change the kernel in this PR, did we?

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb (hunk 5 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2018.03.bb]: What is your proposal for this? Can those patches be done in meta-balena-allwiner?

File: layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample (hunk 2 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/conf/samples/bblayers.conf.sample]: I didn't notice that meta-sunxi was still on on `master`, I checked out to `warrior` and the file is gone. Updated the patchset to reflect this.

**Comment by floion** [comment]
@vicgal let's only have 1 `Changelog-entry` for all the commits that do the submodules updates to warrior. You could have it just in the poky update commit for example.

File: layers/meta-balena-allwinner/conf/machine/bananapi-m1-plus.conf [hunk 7]
```diff
@@ -6,6 +6,7 @@ MACHINEOVERRIDES = "bananapi:${MACHINE}"
 
 include conf/machine/bananapi.conf
 
+PREFERRED_PROVIDER_virtual/kernel = "linux"
+PREFERRED_VERSION_linux = "4.19.76%"
```
Comment (floion) [inline layers/meta-balena-allwinner/conf/machine/bananapi-m1-plus.conf]: I think we could remove this as it will anyway default to 4.19.76 right?

File: layers/meta-balena-allwinner/conf/machine/orangepi-plus2.conf [hunk 8]
```diff
@@ -6,4 +6,5 @@ require conf/machine/include/sun8i.inc
 
 KERNEL_DEVICETREE = "sun8i-h3-orangepi-plus.dtb"
 UBOOT_MACHINE = "orangepi_plus_defconfig"
-PREFERRED_VERSION_linux-mainline = "4.17.3"
+PREFERRED_PROVIDER_virtual/kernel = "linux"
+PREFERRED_VERSION_linux = "4.19.76%"
```
Comment (floion) [inline layers/meta-balena-allwinner/conf/machine/orangepi-plus2.conf]: I think we could remove this as it will anyway default to 4.19.76 right?

File: layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2019.04.bbappend [hunk 9]
```diff
@@ -1,5 +1,5 @@
 UBOOT_KCONFIG_SUPPORT = "1"
-inherit resin-u-boot
+inherit resin-u-boot pythonnative
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-bsp/u-boot/u-boot_2019.04.bbappend]: @vicgal you can also paste the error you got so we know what this commit fixes

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend (hunk 6 above)
Comment (floion) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: Since we are not using this kernel version anymore, I guess we can drop this commit?

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux-4.19/0000-0001-clk-sunxi-ng-Set-maximum-M-1-for-H3-pll-cpux-clock.patch [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+When using M factor greater than 1 system is experiencing
+occasional lockups.
+
+This change was verified to fix lockups with PLL stress
+tester available at https://github.com/megous/h3-firmware.
+
+Note that M factor must not be used outside the kernel
+either, so for example u-boot needs a similar patch.
+---
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-kernel/linux/linux-4.19/0000-0001-clk-sunxi-ng-Set-maximum-M-1-for-H3-pll-cpux-clock.patch]: These patches are missing `Signed-off-by` and `Upstream-status`.

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+KERNEL_EXTRA_ARGS += "LOADADDR=${UBOOT_ENTRYPOINT}"
+
+PV = "4.19.76+git${SRCPV}"
+SRCREV_pn-${PN} = "v4.19.76"
+
+
+SRC_URI = "git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git;protocol=git;branch=linux-4.19.y \
+        "
+S = "${WORKDIR}/git"
```
Comment (floion) [inline layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb]: Where is this recipe taken from?

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux-4.19/0000-0001-clk-sunxi-ng-Set-maximum-M-1-for-H3-pll-cpux-clock.patch (hunk 10 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux-4.19/0000-0001-clk-sunxi-ng-Set-maximum-M-1-for-H3-pll-cpux-clock.patch]: The patches were ported from Armbian. They are about 360. I don't think it's feasible to add a Singed-off-by and Upstream-status in all of them

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend (hunk 6 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: If we would like to go back to kernel 4.14 in the future, for some reason, we will need this commit because the error will still be there if I drop this.

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux-4.19/0000-0001-clk-sunxi-ng-Set-maximum-M-1-for-H3-pll-cpux-clock.patch (hunk 10 above)
Comment (floion) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux-4.19/0000-0001-clk-sunxi-ng-Set-maximum-M-1-for-H3-pll-cpux-clock.patch]: :) ok then, let's leave them like that but please mention the revision (sha) of the armbiam git repo from where you imported them.

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend (hunk 6 above)
Comment (floion) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: Yes we will need this. But let's add it when we need it, I want to avoid stashing stuff we don't use.

Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.14%.bbappend]: Ok. Will drop this commit.

File: layers/meta-balena-allwinner/conf/machine/bananapi-m1-plus.conf (hunk 7 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/conf/machine/bananapi-m1-plus.conf]: Yes, it will default to 4.19.76. Will remove the PREFERRED_VERSION.

File: layers/meta-balena-allwinner/conf/machine/orangepi-plus2.conf (hunk 8 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/conf/machine/orangepi-plus2.conf]: Yes, it will default to 4.19.76. Will remove the PREFERRED_VERSION.

File: layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb (hunk 11 above)
Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb]: It's a new recipe based on the linux_4.14%.bb recipe from the same folder.

Comment (floion) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb]: The warrior branch of meta-sunxi has 4.19.63. Why not use that one?

Comment (floion) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb]: We could use the kernel version in the bsp (4.19.63) and just use the kernel patches from armbian on top of it

Comment (vicgal) [reply layers/meta-balena-allwinner/recipes-kernel/linux/linux_4.19.76.bb]: Some patches don't apply:
(227/344) add-BergMicro-SPI-flashes.patch
| [INFO]: check of .kernel-meta//patches///add-BergMicro-SPI-flashes.patch with "git am" did not pass, trying reduced context.
| [INFO]: Context reduced git-am of .kernel-meta//patches///add-BergMicro-SPI-flashes.patch with "git am" did not work, trying "apply".
| (228/344) add-orangepi-zeroplus2.patch
| [INFO]: check of .kernel-meta//patches///add-orangepi-zeroplus2.patch with "git am" did not pass, trying reduced context.
| [INFO]: Context reduced git-am of .kernel-meta//patches///add-orangepi-zeroplus2.patch with "git am" did not work, trying "apply".
